v1/nix/modules/drv-parts/buildPythonEnv/default.nix

```diff
@@ -0,0 +1,149 @@
+# TODO replace former manualSetupDeps with PEP 518 impl
+{
+  config,
+  lib,
+  drv-parts,
+  ...
+}: let
+  l = lib // builtins;
+  python = config.deps.python;
+  cfg = config.buildPythonEnv;
+  packageName = config.name;
+  metadata = config.lock.content.fetchPipMetadata;
+
+  drvs =
+    l.mapAttrs (
+      name: info:
+        buildDependency {
+          inherit name;
+          inherit (info) version;
+        }
+    )
+    metadata;
+
+  buildDependency = {
+    name,
+    version,
+  }: {config, ...}: {
+    imports = [
+      commonModule
+    ];
+    config = {
+      inherit name version;
+    };
+  };
+
+  commonModule = {config, ...}: {
+    imports = [
+      drv-parts.modules.drv-parts.mkDerivation
+      ../buildPythonPackage
+      ../nixpkgs-overrides
+    ];
+    config = {
+      deps = {nixpkgs, ...}:
+        l.mapAttrs (_: l.mkDefault) {
+          inherit python;
+          inherit
+            (nixpkgs)
+            autoPatchelfHook
+            stdenv
+            ;
+          manylinuxPackages = with nixpkgs.pythonManylinuxPackages; [
+            manylinux1
+            manylinux2010
+            manylinux2014
+          ];
+        };
+      buildPythonPackage = {
+        format = l.mkDefault (
+          if l.hasSuffix ".whl" config.mkDerivation.src
+          then "wheel"
+          else "setuptools"
+        );
+      };
+      mkDerivation = {
+        src = l.mkDefault (l.fetchurl {inherit (metadata.${config.name}) url sha256;});
+        doCheck = l.mkDefault false;
```

Comment:
Tried to do so, but that causes lots of failures because most packages don't include their requirements to run the tests in the default set.

I'd like to leave it like that for now as I am going to add an interface to request "extras" soon. Together with a method to set default for all .drvs in a package, this would allow us to add `{"test", "tests"}` to extras by default if `doCheck`is true, which should solve that problem for most packages in a cleaner way.